evalml/data_checks/outliers_data_check.py

```diff
@@ -42,17 +41,18 @@ def validate(self, X, y=None):
 
         X = infer_feature_types(X)
         X = X.ww.select("numeric")
+        X.ww.drop(X.columns[X.isna().all()].tolist(), inplace=True)
```

Comment:
Is this necessary because `box_plot_dict` was causing issues with fully null columns?

If so, there may be a bug in Woodwork, because I would expect outlier detection be able to [handle fully null columns](https://github.com/alteryx/woodwork/blob/main/woodwork/statistics_utils.py#L389-L402). There wouldn't be any low or high values, so the check on line 53 would not add them to `has_outliers`

Reply:
Yep, when I pass in `nan` data otherwise, I get `TypeError: Cannot calculate box plot statistics for non-numeric column` as the error

Reply:
Ah, this is because we now infer fully null columns as the `Unknown` logical type. 

I believe that the `init_series` call reinferring the logical type as `Unknown` is causing this at the `box_plot_dict` call. Switching to `X.ww[col]` should fix that, since the logical type will remain whatever numeric logical type it is in the DataFrame

Reply:
Ah, in our incorporation of ww 0.5.1, we changed the [nan/null inference](https://github.com/alteryx/evalml/blob/948e3b44dc69f2bd27add1ab9ed24256e2970ea8/evalml/utils/woodwork_utils.py#L67) to be type `Double` rather than `Unknown` in order to handle how we build our components and pipelines. Using `X.ww[col]` caused it to become a Series object rather than a ww DataColumn obj, which then raised the `AttributeError: 'Series' object has no attribute 'box_plot_dict'` error here

Reply:
Not sure I'm following.

Both `ww.init_series(X[col])` and `X.ww[col]` should return a Series that has Woodwork initialized on it. If Woodwork was not initialized on a Series and you did `col_series.ww.box_plot_dict()`, the error would be `WoodworkNotInitError: Woodwork not initialized for this Series. Initialize by calling Series.ww.init`, and if the attribute did not exist in Woodwork, it would look more like `AttributeError: Woodwork has no attribute 'doesnt_exist'`.

The attribute error you're showing seems more like what you get if you try and access a Woodwork attribute not through the Woodwork accessor (so `df.box_plot_dict()` gets this exact error, when `df.ww.box_plot_dict()` would work)

Reply:
Ah yeah, I was able to get it to work! It was a mistake on my end. Does the implementation look better to you?

Reply:
Looks great! Love the chaining with `X.ww[col].ww.box_plot_dict()`.

Reply:
Thanks for the feedback and help!

Reply:
So happy to help! And I'd love to hear if you have any feedback on if there's anything we can be doing to make Woodwork's `ww` usage a little easier to understand.